src/test/java/net/ripe/rpki/commons/validation/X509ResourceCertificateParentChildValidatorTest.java

```diff
@@ -225,6 +237,117 @@ public void shouldRejectInheritedResourcesForSelfSignedCertificate() {
         assertTrue(result.hasFailures());
     }
 
+    @Property
+    public void validParentChildSubResources(List<@From(IpResourceGen.class) IpResource> parentResources, int childResourceCount) {
+        if (parentResources.isEmpty()) {
+            return;
+        }
+
+        final IpResourceSet parentResourceSet = new IpResourceSet(parentResources);
+
+        // some part of the parent resources become child
+        final IpResourceSet childResourceSet = new IpResourceSet(
+            parentResources.subList(0, Math.abs(childResourceCount) % parentResources.size()));
+        if (childResourceSet.isEmpty()) {
+            return;
+        }
+
+        ValidationResult result = validateParentChildPair(parentResourceSet, childResourceSet);
+        assertFalse(result.hasFailures());
+    }
+
+    @Property
+    public void validParentChildOverClaiming(List<@From(IpResourceGen.class) IpResource> parentResources,
+                                             int childResourceCount,
+                                             List<@From(IpResourceGen.class) IpResource> extraChildResources) {
+        if (parentResources.isEmpty()) {
```

Comment:
```suggestion
        assumeThat(parentResources.isEmpty(), is(false));
```

If this is more readable.

Reply:
Yeah, makes sense.